vcr/pe/presentation_definition.go

```diff
@@ -47,16 +53,39 @@ type Candidate struct {
 // - number, boolean, array and string JSON schema types
 // - Submission Requirements Feature
 // It doesn't do the credential search, this should be done before calling this function.
-// The PresentationDefinition.Format should be altered/set if an envelope defines the supported format before calling.
-// The resulting PresentationSubmission has paths that are relative to the matching VCs.
-// The PresentationSubmission needs to be altered so the paths use "path_nested"s that are relative to the created VP.
+// The given PresentationContext is used to set the correct vp index in the nested paths and to alter the given PresentationSubmission.
+// It assumes this method is used for OpenID4VP since other envelopes require different nesting.
 // ErrUnsupportedFilter is returned when a filter uses unsupported features.
 // Other errors can be returned for faulty JSON paths or regex patterns.
-func (presentationDefinition PresentationDefinition) Match(vcs []vc.VerifiableCredential) (PresentationSubmission, []vc.VerifiableCredential, error) {
+func (presentationDefinition PresentationDefinition) Match(context PresentationContext, vcs []vc.VerifiableCredential) ([]vc.VerifiableCredential, error) {
```

Comment:
weird public-facing API; what to does the caller need to do with "Index"? This should probably be an internal shelper struct. And this function should create the submission, in my opinion (or there should be a  separate `PresentationDefinition.CreateSubmission` which does this, potato/potato). Now callers have to set the right `DefinitionId`, while this is part of creating the submission.

Reply:
I'll create an `AddMatch` to a PresentationSubmission. That way the submission can also handle the index and single vs plural paths.